File: 452.java
import java.util.*;
class Problem452 {
    public int findMinArrowShots(int[][] points) {
        if (points.length == 0) return 0;
        Arrays.sort(points, (a,b)->Integer.compare(a[1], b[1]));
        for (int[] point: points) System.out.println(Arrays.toString(point));
        
        int arrows = 1, last = points[0][1];
        for (int i = 1; i < points.length; i++){
            if (points[i][0] > last){
                arrows++;
                last = points[i][1];
            }
        }
        return arrows;
    }
}

File: 452.py
class Solution(object):
    def findMinArrowShots(self, points):
        points = sorted(points, key = lambda x: x[1])
        arrows = 1
        placementArrow = points[0][1]
        for index in range(1, len(points)):
            if points[index][0] > placementArrow:
                placementArrow = points[index][1]
                arrows += 1
        return arrows